### PR TITLE
Fixes #29: Make getDeclarationByName work with type expressions

### LINKED ISSUE #29: Type expressions aren't returned by getDeclarationByName
Reported in https://github.com/raml-org/webapi-parser/issues/28

Code to test with:
```js
const wap = require('webapi-parser').WebApiParser

const ramlStr = `
  #%RAML 1.0 Library
  types:
    Foo:
      type: object
      properties:
        foo: string
    Bar:
      type: object
      properties:
        bar: string
    FooBar: Foo | Bar
    BarBaz: [Foo, Bar]
    Foos: Foo[]
    FoosBars: (Foo | Bar)[]
    strings: string[]
    strings2: string[][]
`

async function main () {
  const model = await wap.raml10.parse(ramlStr)

  const foo = model.getDeclarationByName('Foo');
  console.log("foo:", foo);
  
  const bar = model.getDeclarationByName('Bar');
  console.log("bar:", bar);
  
  const foobar = model.getDeclarationByName('FooBar');
  console.log("foobar:", foobar);
}

main()
```

### PR DESCRIPTION
**Pull request by postatum**
Fixes #29
---
Things that need to be done after changes of this PR are released: 
* generate js and java docs in master
* describe breaking changes in a release description
---
### Breaking changes
This PR presents potentially breaking changes for Java users. In particular `WebApiBaseUnit.getDeclarationByName` now returns an instance of `amf.client.model.domain.AnyShape` instead of `amf.client.model.domain.NodeShape`. Note that `NodeShape` is a subclass of `AnyShape` as [API docs state](https://raml-org.github.io/webapi-parser/js/classes/_amf_client_js_.model.domain.anyshape.html).

Considering this, you might need to explicitly convert an instance of `AnyShape` returned by `WebApiBaseUnit.getDeclarationByName` to its subclasses to access subclass-specific properties.

E.g. to access [UnionShape.anyOf](https://raml-org.github.io/webapi-parser/js/classes/_amf_client_js_.model.domain.unionshape.html#anyof) property you would have to do:
```java
UnionShape user = (UnionShape) model.getDeclarationByName("User");
System.out.println(user.anyOf());
```

JS users should face no breaking changes.

### CODE REVIEW

**Comment by postatum** [comment]
This PR needs to be updated a little. `getDeclarationByName` doesn't return scalar types. E.g. 
```
  CatBreed:
    type: string
```

File: build.sbt [hunk 1]
```diff
@@ -27,8 +27,8 @@ val settings = Common.settings ++ Common.publish ++ Seq(
   credentials ++= Common.credentials(),
   libraryDependencies ++= Seq(
     "org.scalatest"     %%% "scalatest"       % "3.0.5" % "test",
-    "com.github.amlorg" %%% "amf-webapi"      % "3.2.1",
-    "com.github.amlorg" %%% "amf-validation"  % "3.2.1"
+    "com.github.amlorg" %%% "amf-webapi"      % "3.2.2",
```
Comment (jstoiko) [inline build.sbt]: latest is [3.3.0](https://github.com/aml-org/amf/releases/tag/3.3.0)

Comment (postatum) [reply build.sbt]: @jstoiko can we make this upgrade in a separate PR?
Release 3.3.0 adds breaking changes (section "Interface Changes") which may break things in webapi-parser code and/or examples. There will be too much in this PR if we fix migration issues here.

**Comment by postatum** [comment]
Done reworking the PR. `WebApiBaseUnit.getDeclarationByName` should now return all valid RAML 1.0 type declarations.

File: build.sbt (hunk 1 above)
Comment (postatum) [reply build.sbt]: Also see https://github.com/aml-org/amf/issues/542

**Pull request merged**